packages/medusa/src/services/event-bus.ts

```diff
@@ -298,18 +334,21 @@ export default class EventBusService {
       )
       const jobs = await stagedJobRepo.find(listConfig)
 
-      await Promise.all(
-        jobs.map((job) => {
-          this.queue_
-            .add(
-              { eventName: job.event_name, data: job.data },
-              { jobId: job.id, ...job.options }
-            )
-            .then(async () => {
-              await stagedJobRepo.remove(job)
-            })
-        })
-      )
+      if (!jobs.length) {
+        await sleep(3000)
+        continue
```

Comment:
question: Why not use `return`? `continue` is for breaking loop iterations
```suggestion
        return
```

Reply:
Continue will stop the current iteration and move to the next one. Return will stop the method execution

Reply:
Ah yeah sorry. I forgot this was nested within a `while( ... ) {`